modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/SubscriptionDLAppHelperLocalServiceWrapper.java

```diff
@@ -90,6 +91,20 @@ public void deleteFolder(Folder folder) throws PortalException {
 			folder.getFolderId());
 	}
 
+	@Override
+	public void getEntryURL(FileEntry fileEntry, ServiceContext serviceContext)
+		throws PortalException {
+
+		super.getEntryURL(fileEntry, serviceContext);
+
+		String friendlyURL = _getFriendlyURL(
+			fileEntry.getFileEntryId(), serviceContext);
+
+		if (Validator.isNotNull(friendlyURL)) {
+			serviceContext.setAttribute("entryURL", friendlyURL);
+		}
+	}
+
```

Comment:
This looks very strange. Why is the method called `get*` when it doesn't "get" anything? Also, why is this modifying the service context so low on the call chain? This feels wrong, like the logic is at the wrong place.

Reply:
Changed.

Can you do, at least a quick check in order to provide a emergency hotfix to a customer?